models/sipnet/R/write.restart.SIPNET.R

```diff
@@ -1,8 +1,37 @@
-write.restart.SIPNET<- function(nens,outdir,run.id,time,settings,prior,analysis){
-  for(i in 1:nens){   
-    file.rename(file.path(outdir,run.id[[i]],"sipnet.out"),file.path(outdir,run.id[[i]],paste0("sipnet.out",time[t])))
-    file.remove(file.path(settings$rundir,run.id[[i]],"sipnet.clim"))
-    do.call(my.write.config,args=list(defaults,list(pft=prior[i,],env=NA),
-                                      settings, run.id[[i]],inputs = settings$run,IC=analysis[i,-1]))   
+write.restart.SIPNET<- function(out.dir, runid, time, settings, analysis.vec,
+                                RENAME = TRUE, PLOT = FALSE, variables,
+                                sample_parameters = FALSE,
+                                trait.values = NA, my.write.config, met){
+ 
+  if(RENAME == TRUE) {
+    file.rename(file.path(outdir,runid,"sipnet.out"),
+                file.path(outdir,runid,paste0("sipnet.out",time)))
+    system(paste("rm",file.path(rundir,runid,"sipnet.clim")))
+  }else{
+    print(paste("Files not renamed -- Need to rerun year",time,"before next time step"))
   }
+   #file.remove(file.path(settings$rundir,runid,"sipnet.clim"))
+  
+  settings$run$start.date <- paste0(time + 1,"/01/01")
+  settings$run$end.date <- paste0(time + 1,"/12/31")
+    
+  if(sample_parameters == TRUE){
+    load(file.path(settings$outdir, "samples.Rdata"))
+    do.call(my.write.config, args = list(defaults = NULL, trait.values = trait.values,
```

Comment:
parameter sampling should be isolated to the start of the ensemble, and then ensemble members should carry their parameters forward with them, not read them from some general file.  That said, when you do sample parameters you need to make sure to get the correct posterior ID's from the settings file. People will be running SDA after PDA and we want them to be using those posteriors.